File: api/src/main/java/dev/obscuria/fragmentum/util/event/Event.java
package dev.obscuria.fragmentum.util.event;

import org.apache.commons.lang3.NotImplementedException;

@SuppressWarnings("unused")
public final class Event<T>
{
    public EventToken register(T listener)
    {
        return register(1000, listener);
    }

    public EventToken register(int priority, T listener)
    {
        throw new NotImplementedException();
    }

    public void unregister(EventToken token)
    {
        throw new NotImplementedException();
    }

    public void broadcast(EventHandler<T> handler)
    {
        throw new NotImplementedException();
    }
}

File: api/src/main/kotlin/dev/obscuria/fragmentum/tools/event/Event.kt
@file:Suppress("unused", "UNUSED_PARAMETER", "RedundantNullableReturnType", "MemberVisibilityCanBePrivate")

package dev.obscuria.fragmentum.tools.event

class Event<T> {

    infix fun register(listener: T): EventToken = register(1000, listener)

    fun register(priority: Int, listener: T): EventToken = error("API stub")

    infix fun unregister(token: EventToken): Unit = error("API stub")

    infix fun broadcast(handler: EventHandler<T>): Unit = error("API stub")
}